pgrx/src/lib.rs

```diff
@@ -327,3 +323,16 @@ pub(crate) enum Utf8Compat {
     /// An "extended ASCII" encoding, so we're fine if we only touch ASCII
     Ascii,
 }
+
+#[doc(hidden)]
```

Comment:
This is visibly called by a binary we template into their project directory, and they can read it without doing something like `cargo expand`, so it should have documentation. We should explain this macro is part of schema generation and it should not be called outside of the schema-gen binary, because it serves as an entry point for inserting the code our proc macros expand to generate the schema, which will be arbitrary code.

Reply:
For example:

```suggestion
/// Entry point for cargo-pgrx's schema generation so that PGRX's framework can
/// generate SQL for its types and functions and topographically sort them into
/// an order Postgres will accept. Typically written by the `cargo pgrx new`
/// template, so you probably don't need to worry about this.
```